src/main/scala/com/campudus/tableaux/bus/TableauxBusMod.scala

```diff
@@ -1,80 +1,80 @@
-      controller.changeColumn(getInfo[Long](msg, "tableId"), getInfo[Long](msg, "column"), optName, optOrd, optKind)
-    }
-    case "deleteTable" => getAsyncReply(EmptyReturn)(controller.deleteTable(getInfo[Long](msg, "tableId")))
-    case "deleteColumn" => getAsyncReply(EmptyReturn)(controller.deleteColumn(getInfo[Long](msg, "tableId"), getInfo[Long](msg, "columns")))
-    case "deleteRow" => getAsyncReply(EmptyReturn)(controller.deleteRow(getInfo[Long](msg, "tableId"), getInfo[Long](msg, "rows")))
-    case _ => throw new IllegalArgumentException("Unknown action")
-  }
-
-  private def getInfo[A](msg: Message[JsonObject], name: String): A = name match {
-    case "tableName" | "tableId" => get[A](msg, name)
-    case "columns" | "rows" =>
-      val column = get[JsonArray](msg, name).get[JsonObject](0)
-      get[A](column, "id")
-    case "column" | "row" =>
-      val cell = get[JsonArray](msg, "cells").get[JsonObject](0)
-      val column = get[JsonObject](cell, name)
-      get[A](column, "id")
-  }
-
-  private def get[A](msg: Message[JsonObject], name: String): A = msg.body.getField[A](name)
-
-  private def get[A](msg: JsonObject, name: String): A = msg.getField[A](name)
-
-  private def getAsyncReply(reType: ReturnType)(f: => Future[DomainObject]): AsyncReply = AsyncReply {
-    f map { d => Ok(d.toJson(reType)) } recover {
-      case ex @ NotFoundInDatabaseException(message, id) => Error(message, s"errors.database.$id")
-      case ex @ DatabaseException(message, id) => Error(message, s"errors.database.$id")
-      case ex @ NoJsonFoundException(message, id) => Error(message, s"errors.json.$id")
-      case ex @ NotEnoughArgumentsException(message, id) => Error(message, s"errors.json.$id")
-      case ex @ InvalidJsonException(message, id) => Error(message, s"error.json.$id")
-      case ex: Throwable => Error("unknown error", "errors.unknown")
-    }
-  }
-
+package com.campudus.tableaux.bus
+
+import com.campudus.tableaux._
+import com.campudus.tableaux.controller.TableauxController
+import com.campudus.tableaux.database.structure.{ReturnType, EmptyReturn, GetReturn, SetReturn}
+import com.campudus.tableaux.database.{DomainObject, _}
+import com.campudus.tableaux.helper.HelperFunctions._
+import com.campudus.tableaux.helper.{HelperFunctions, StandardVerticle}
+import org.vertx.scala.core.eventbus.Message
+import org.vertx.scala.core.json.{JsonArray, JsonObject}
+import org.vertx.scala.mods.ScalaBusMod
+import org.vertx.scala.mods.replies._
+import org.vertx.scala.platform.Verticle
+
+import scala.concurrent.Future
+
+class TableauxBusMod(val verticle: Verticle, val database: DatabaseConnection) extends ScalaBusMod with DatabaseAccess with StandardVerticle {
+
+  val controller = new TableauxController(verticle, database)
+
+  override def receive: Message[JsonObject] => PartialFunction[String, BusModReceiveEnd] = msg => {
+    case "reset" => getAsyncReply(SetReturn)(controller.resetDB())
```

Comment:
We could create shorter variables for `getAsyncResult(XXX)` (only the first `()`). Like:
```
val setAsync = getAsyncReply(SetResult)
val getAsync = getAsyncResult(GetReturn)
val emptyAsync = getAsyncResult(EmptyReturn)
```

Then we could have `case "reset" => setAsync(controller.resetDB())`, etc..